### ISSUE TITLE
Megatron DDP strategy based MegatronLossReduction

### ISSUE THREAD

**Issue by piyushumate** [NVIDIA/NeMo#13630]
Is there a sample reference/example on how the MegatronLossReduction should look in case of Megatron DDP?

**Comment by maanug-nv** [comment]
Hi @piyushumate , by Megatron DDP, are you referring to the MegatronStrategy (in `nemo.lightning.pytorch.strategies`) or the `DistributedDataParallel` wrapper from `megatron-core` (`megatron.core.distributed.distributed_data_parallel`) ?

**Comment by maanug-nv** [comment]
If you are referring to the former, I'd recommend looking at this quickstart [guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo-2.0/index.html#nemo-2-quickstart-api) for Nemo 2.0 , for details on how to set up the Trainer, strategy, and model. If you take a look at the `GPTModel` [here](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/llm/gpt/model/base.py#L711-L732),  you can see that the GPTModel creates a `MaskedTokenLossReduction` for its training and validation loss reductions. This will be used by the `MegatronParallel` wrapper. Are you trying to use a `MegatronLossReduction` with a customized model and `MegatronStrategy`?

**Comment by maanug-nv** [comment]
Hi @piyushumate , did the above work for you? Let me know if you are still looking for an example or if we can close this request.
